File: ag-ui/integrations/aws-strands/typescript/examples/server/api/agentic-chat-reasoning.ts
/**
 * Agentic Chat with Reasoning example for AWS Strands (TypeScript).
 *
 * Demonstrates reasoning/thinking event streaming. When the underlying model
 * supports extended thinking, the adapter emits REASONING_* events that the
 * frontend can display as a "thinking" indicator.
 */

import { Agent } from "@strands-agents/sdk";
import { StrandsAgent } from "@ag-ui/aws-strands";
import { createStrandsApp } from "@ag-ui/aws-strands/server";
import { createModel } from "../model-factory";

async function main(): Promise<void> {
  const strandsAgent = new Agent({
    model: await createModel({ reasoning: true }),
    systemPrompt: `
    You are a helpful assistant that thinks through problems step by step.
    When the user greets you, always greet them back. Your greeting should always start with "Hello".
    Your greeting should also always ask (exact wording) "how can I assist you?"
    When reasoning about a problem, break it down into clear steps before answering.
  `,
  });

  const aguiAgent = new StrandsAgent({
    agent: strandsAgent,
    name: "agentic_chat_reasoning",
    description:
      "Conversational Strands agent with reasoning/thinking event streaming",
  });

  const app = await createStrandsApp(aguiAgent, { path: "/" });
  const port = Number(process.env.PORT ?? 8000);
  app.listen(port, () => {
    console.log(`Listening on http://localhost:${port}`);
  });
}

void main();

File: ag-ui/integrations/agno/python/examples/server/api/agentic_chat_reasoning.py
"""Agentic Chat with Reasoning — Uses reasoning models (o4-mini) that show their thinking process."""

from agno.agent.agent import Agent
from agno.models.openai import OpenAIResponses
from agno.os import AgentOS
from agno.os.interfaces.agui import AGUI

# o4-mini is a reasoning model that exposes its thinking process via reasoning tokens
# Use OpenAIResponses with reasoning_effort + agent reasoning=True to emit reasoning events
agent = Agent(
    model=OpenAIResponses(
        id="o4-mini",
        reasoning_effort="high",
        reasoning_summary="auto",
    ),
    reasoning=True,
    description="You are a helpful AI assistant with deep reasoning capabilities.",
    instructions=[
        "Think step by step through complex problems.",
        "Explain your reasoning clearly.",
    ],
    markdown=True,
)

agent_os = AgentOS(agents=[agent], interfaces=[AGUI(agent=agent)])

app = agent_os.get_app()
